packages/vuetify/src/components/VSelect/VSelect.tsx

```diff
@@ -137,7 +137,7 @@ export const VSelect = genericComponent<new <T>() => {
         const index = selections.value.findIndex(selection => selection.value === item.value)
 
         if (index === -1) {
-          model.value.push(item.value)
+          model.value = model.value.slice().concat([item.value])
         } else {
           model.value = selected.value.filter(selection => selection !== item.value)
```

Comment:
Unrelated, but if we already have the index can't we skip the filter to save a loop?
```suggestion
          const value = [...selected.value]
          value.splice(index, 1)
          model.value = value
```